packages/slice-machine/server/src/api/screenshots/generate.ts

```diff
@@ -48,28 +47,23 @@ export async function generateScreenshotAndRemoveCustom(
 export async function generateScreenshot(
   env: BackendEnvironment,
   libraryName: string,
-  sliceName: string
+  sliceName: string,
+  variationId: string
 ): Promise<ScreenshotResults> {
   // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
   const slice = IO.Slice.readSlice(
     NodeUtils.CustomPaths(env.cwd).library(libraryName).slice(sliceName).model()
   );
 
-  const variationIds: VariationSM["id"][] = slice.variations.map((v) => v.id);
-
-  const promises: Promise<ScreenshotUI>[] = variationIds.map(
+  const promises: Promise<ScreenshotUI>[] = [variationId].map(
```

Comment:
We don't need all the logic looping on the variations anymore.

Reply:
I think when me an Mathieu wrote it we opted to leave it like this since there was a lot of work in there and we weren't sure if soon we'd have a feature like "update all variations" that would need to use it again.

If you don't think it's needed then I'll refactor the whole thing

Reply:
It shouldn't be needed in this iteration.
From what I know they don't want automatic screenshots for multiple variations anymore.